src/table/src/metadata.rs

```diff
@@ -309,30 +278,33 @@ impl TableMeta {
         self.set_table_options(&requests)
     }
 
-    /// Creates a [TableMetaBuilder] with modified column inverted index.
-    fn change_column_modify_inverted_index(
+    fn set_indexes(
         &self,
         table_name: &str,
-        column_name: &str,
-        value: bool,
+        requests: &[SetIndexOption],
     ) -> Result<TableMetaBuilder> {
         let table_schema = &self.schema;
-        let mut meta_builder = self.new_meta_builder();
-
-        let mut columns: Vec<ColumnSchema> =
-            Vec::with_capacity(table_schema.column_schemas().len());
+        let mut set_index_options = HashMap::with_capacity(requests.len());
+        for request in requests {
+            let column_name = request.column_name();
+            table_schema
+                .column_index_by_name(column_name)
+                .with_context(|| error::ColumnNotExistsSnafu {
+                    column_name,
+                    table_name,
+                })?;
+            set_index_options.insert(column_name, request);
+        }
```

Comment:
Is it possible that requests contain multiple requests for the same column name?

Reply:
resolved in [ae4d8a4](https://github.com/GreptimeTeam/greptimedb/pull/6487/commits/ae4d8a4740787a4261c7b3f88152d88aac0e2df4)